src/moscot/backends/ott/solver.py

```diff
@@ -192,7 +214,10 @@ def _prepare(
         )
         if cost_matrix_rank is not None:
             geom = geom.to_LRCGeometry(rank=cost_matrix_rank)
-        self._problem = linear_problem.LinearProblem(geom, **kwargs)
+        if xy.is_graph:
+            a = jnp.concatenate((a, jnp.zeros_like(self._b)), axis=0)
+            b = jnp.concatenate((jnp.zeros_like(self._a), b), axis=0)
```

Comment:
`self._a` and `self._b` are used only here or also somewhere downstream? Because If it's only here, why not getting the shape of the left and right marginal from the geometry `geom` ?

Reply:
we cannot get the shape from the `geom` because the `geom` is of shape `[n+m, n+m]`